lib/uefi/boot_service_provider.cpp

```diff
@@ -0,0 +1,380 @@
+
+#include "boot_service_provider.h"
+#include "boot_service.h"
+#include "kernel/vm.h"
+#include "lib/dlmalloc.h"
+#include "types.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+EfiStatus unload(EfiHandle handle) { return SUCCESS; }
+
+bool guid_eq(const EfiGuid *a, const EfiGuid *b) {
+  return memcmp(a, b, sizeof(*a)) == 0;
+}
+
+bool guid_eq(const EfiGuid *a, const EfiGuid &b) {
+  return memcmp(a, &b, sizeof(*a)) == 0;
+}
+
+vmm_aspace_t *get_address_space() {
+  static vmm_aspace_t *aspace = nullptr;
+  if (aspace == nullptr) {
+    auto err = vmm_create_aspace(&aspace, "linux_kernel", 0);
+    if (err) {
+      printf("Failed to create address space for linux kernel %d\n", err);
+      return nullptr;
+    }
+    vmm_set_active_aspace(aspace);
```

Comment:
This will trash whatever the existing aspace was, which is probably not a huge concern at the moment, but we should perhaps save a copy of the existing one?

Reply:
Done